src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -19,92 +19,135 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Configuration;
 using System.Diagnostics;
-using System.Web;
 using System.IO;
 using System.Text;
+using System.Web;
 
 public class PullRequestHandler : IHttpHandler
 {
-    private const string _gitExecutablePath = @"C:\Program Files\Git\cmd\git.exe";
-    private static string _gitRepoPath;
-    private const int _gitExecutionTimeoutMilliseconds = 100000;
-
-    public void ProcessRequest(HttpContext context)
-    {
-        // Check for prerequisites needed before this script can execute
+    private string _gitExecutablePath;
+    private string _gitRepositoriesBasePath;
+    private TimeSpan _gitExecutionTimeout;
 
-        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+    private bool _configurationRead = false;
 
-        StringBuilder result = new StringBuilder();
-        bool success = CheckPrerequisites(result);
+    private const string _gitExecutablePathConfigurationKeyString = "GitExecutablePath";
+    private const string _gitRepositoriesBasePathConfigurationKeyString = "GitRepositoriesBasePath";
+    private const string _gitExecutionTimeoutConfigurationKeyString = "GitExecutionTimeout";
 
-        if (!success)
-        {
-            // CheckPrerequisites failed, this is a server issue
-            context.Response.StatusCode = 500;
-            context.Response.Write(HttpUtility.HtmlEncode(result));
-            return;
-        }
+    private const string _gitRepoPathTemplate = @"{0}\wcf{1}";
 
-        success = CleanupMergedBranches(result);
+    public void ProcessRequest(HttpContext context)
+    {
+        StringBuilder result = new StringBuilder();
 
+        // Read configuration
+        bool success = ReadConfiguration(result);
         if (!success)
         {
-            // CleanupMergedBranches failed, this is a server issue
+            // ReadConfiguration failed, this is a server issue 
             context.Response.StatusCode = 500;
-            context.Response.Write(HttpUtility.HtmlEncode(result));
+            context.Response.Write(HttpUtility.HtmlEncode(result.ToString()));
             return;
         }
 
-        // Parameter verification
+        // Check inputs 
+        // We expect one of the following to be specified: 
+        // - A numerical repo ID
+        // - A branch name OR a PR number (not both)
+        //
+        // We will use Git to validate and verify that the branch exists if specified
 
         success = false;
 
+        string repoString = context.Request.QueryString["repo"];
```

Comment:
The parameter `repo` is the API surface of PRService. In the perspective of PRService, it asks callers to identify themselves so the PRService can differentiate. Although we use different repo's behind the scene, we should not expose it on the API surface. Can you change this to `id` please?